pages/wizard/product_selection.py

```diff
@@ -39,3 +41,15 @@ def click_next(self):
     # there is no back button on this page:
     def click_back(self):
         return None
+
+    @property
+    def is_the_current_page(self):
+        WebDriverWait(self.selenium, self.timeout).until(
+            EC.visibility_of(self.rhev_checkbox),
+            "Expected element not present on Select Products Page")
+        return True
+
+    def cancel_deployment(self):
```

Comment:
The thing is this is stateful.   If you haven't gotten to the insights page, if you go back to the product selection page then cancel just takes you back to the deployments page.   However once you have gotten to the insights page, if you go back to the product selection page, cancel will give you the modal frame that lets you:

* exit and delete
* exist and save
* continue working

I don't know how we should handle that, because right now we are pretty stateless.   Maybe there is a way to tell from the page, but I don't know how at this point.